### ISSUE TITLE
TOTK freezing then crashing after coming out of weapon selection menu

### ISSUE THREAD

**Issue by lavinsky** [yuzu-emu/yuzu#10274]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Affected Build(s)

Early Access 3588, Mainline 1430

### Description of Issue

Game freezes briefly after coming out of weapon selection menu, sometimes the game will recover and keep running. If you do it many times in a row yuzu will crash.

### Expected Behavior

Not freezing when coming out of weapon selection menu. Never crashing.

### Reproduction Steps

Get in game, keep pressing and releasing dpad right to come in and out of weapon selection menu

### Log File

[yuzu_log.txt](https://github.com/yuzu-emu/yuzu/files/11469961/yuzu_log.txt)

### System Configuration

CPU: Ryzen 5 5500
GPU/Driver: AMD RX 570 / Adrenalin 23.3.1 (as suggested by the latest progress report, 23.4.3 doesn't solve the issue)
RAM: 16gb
OS: Windows 11 22H2

**Comment by Squall-Leonhart** [comment]
do you have yuzu fix applied still?, both the sdk and yuzufix from the 1.x files need to be removed at this point.

**Comment by liamwhite** [comment]
I can't reproduce this at all. Menuing works without hitching or crashing for me.

**Comment by hydall** [comment]
Same issue.

**System Configuration**
CPU: i5-6300HQ
GPU: GTX960M 2GB, driver is the latest available
RAM: 8GB
OS: Windows 10 21H2

**Early Access 3588**

What I have noticed is that it happens when you don't have **"Decode ASTC textures asynchronously (Hack)"**. The issue goes away when you do enable it, but, for me, the consequences of it is visual artifacts such as random textures on in-world objest and HUD elements (weapon carousel, map while loading, Link's body etc). Probably my old platform to blame. Try that fix for yourself.

Speaking of **"Accelerate ASTC texture decoding"** - I haven't noticed any effect on the issue after enabling it.

**Comment by liamwhite** [comment]
https://github.com/yuzu-emu/yuzu/assets/9658600/785d2289-fb5d-4894-acd7-9849bcd84d87

Is there something else I am supposed to be doing?

**Comment by hydall** [comment]
> menu.mp4 
> Is there something else I am supposed to be doing?

Do you have? **"Decode ASTC textures asynchronously (Hack)"** enabled? If so, then you won't have that issue. If no, then it's hard to tell what's the cause of that.

**Comment by liamwhite** [comment]
No. I have "Accelerate ASTC texture decoding" enabled (to use the compute decoder) and "Decode ASTC textures asynchronously (Hack)" disabled.

**Comment by Rusxaker** [comment]
Same issue.

System Configuration
CPU: i7-4790K
GPU: GTX 970 4GB, Driver is 472.12 (on the latest version of the drivers the game crashes when using Vulkan API)
RAM: 16GB
OS: Windows 10 21H2

Early Access 3588

I have a problem only when using Vulkan, when using OpenGL there is no such problem, but the game is less stable.

**Comment by Squall-Leonhart** [comment]
The only vulkan issues i'm aware of post 472 involve end user systems that have got unmaintained out of date vulkan layers installed, such as the twitch studio layer, or an out of date mirillis action layer.

**Comment by helkin-jurgles** [comment]
i have the same issue too. there is a freeze whenever the game "blurs" opening menu switching weapon switching rune opening the quick item select

System Configuration
CPU: i5-7500
GPU: GTX 1050ti 4GB, Driver is nvidia-driver-525 
RAM: 8GB
OS: Linux Mint

Early Access 3588

i thought i might be the ram causing this but it seems to not cause a problem on opengl. not using opengl as it takes my fps down from stable 30. using the 1.1 update and 30pfs patch and problem occurs even with them off and shaders deleted

**Comment by Squall-Leonhart** [comment]
> but it seems to not cause a problem on opengl.

OpenGL driver can use Available VRAM+System Shared (PCIE Aperture)
Vulkan can only use Available VRAM, since someones poor decision when the specification was drafted didn't allow for the possibility of overcommit.

**Comment by helkin-jurgles** [comment]
i am sorry,i am a bit dumb but is this a issue with the lack of ram or the lack of vram and if so is there any fix possible other than to upgrae

**Comment by lavinsky** [comment]
Finally got home so I could try to find out more about the issue. Throughout the testing I have not been using any mods since I seem to get worse performance when using any of them, that includes the 30fps mod and the dynamic fps one. Issue still happens with mods enabled. All testing was conducted on EA3588

1. When the emulator crashes the driver crashes along with the game.
2. Activating extended ram layout on the settings seems to alleviate the issue but not fix it entirely.
3. When getting in and out of the menu textures seem to be loading then unloading, vram usage raises and drops accordingly.
4. When coming out of the menu everything blinks very briefly and vram usage raises again.
5. Async ASTC decode (hack) causes texture corruption
6. Disabling "Accelerate ASTC texture decoding" (ASTC compute shader) seems to fix the crashes but not the long freezes (more testing needed)
7. @liamwhite Issue seem to affect low vram gpus (4gb or less). Could that be why you can't reproduce it?

ASTC compute shader enabled video:
https://github.com/yuzu-emu/yuzu/assets/16194440/be7675ca-d923-4f10-bf9e-0915685bdc9c

ASTC compute shader DISABLED:
https://github.com/yuzu-emu/yuzu/assets/16194440/751afb79-1f3d-4121-907f-263f63fb9943

**Comment by Squall-Leonhart** [comment]
The fix in #10286 may help, if its vram related.

**Comment by Rusxaker** [comment]
> The fix in #10286 may help, if its vram related.

no, the problem remains on Yuzu mainline 1432 and Yuzu EA 3589.

**Comment by Squall-Leonhart** [comment]
> no, the problem remains on Yuzu mainline 1432 and Yuzu EA 3589.

Theres an elephant in the room that prevents anyone from trying to assist you on those drivers, install and try 531.86.

**Comment by Rusxaker** [comment]
> 

yuzu on Vulkan crashing with any drivers higher than 512.95, this problem a NVIDIA Drivers. #8887

**Comment by Rusxaker** [comment]
> > no, the problem remains on Yuzu mainline 1432 and Yuzu EA 3589.
> 
> Theres an elephant in the room that prevents anyone from trying to assist you on those drivers, install and try 531.86.

yuzu on Vulkan crashing with any drivers higher than 512.95, this problem a NVIDIA Drivers. https://github.com/yuzu-emu/yuzu/issues/8887

**Comment by Squall-Leonhart** [comment]
Notice the fact its closed and confirmed fixed in the closing comment.

**Comment by Rusxaker** [comment]
> Notice the fact its closed and confirmed fixed in the closing comment of #9146

It is written that the problem is solved in NVIDIA Drivers 527.37, but I installed 527.37 and 531.79 and 531.83 and in all cases yuzu on Vulkan crashes.

**Comment by Squall-Leonhart** [comment]
Ah, The blog says its fixed for Pascal+

Maxwell can upgrade to 512.95 instead of staying on 472.12

**Comment by Rusxaker** [comment]
> Ah, The blog says its fixed for Pascal+
> 
> Maxwell can upgrade to 512.95 instead of staying on 472.12

On drivers above 472.12 and up to 512.95, the emulator does not crash but has a problem with black textures. [Black Textures problem on Yuzu Vulkan TOTK](https://www.reddit.com/r/NewYuzuPiracy/comments/13b3y3i/black_textures_problem_on_yuzu_vulkan_totk/)

**Comment by lavinsky** [comment]
Issue persists on EA3589 and mainline 1433.

**Comment by liamwhite** [comment]
Reproduced the issue on an RX 560 4GB. Investigating now.

**Comment by liamwhite** [comment]
This bug seems specific to older AMD cards. I can't get it to happen on newer AMD or with Nvidia.

**Comment by lavinsky** [comment]
I've been testing, that's right, crashing seems to have stopped or at least reduced in 3589 when accelerated ASTC is enabled. Freezing persists, happens no matter which combination of settings I use.

**Comment by lavinsky** [comment]
Downloaded Ubuntu to test if it still happens on MESA drivers, confirmed it happens

**Comment by DapperCodErr** [comment]
Hello, I have the exact same issue where returning from Links inventory would freeze the game. And I'm getting the following errors in the log:

> [ 563.072048] HW.Memory \<Error\> core\memory.cpp:operator ():239: Unmapped ReadBlock @ 0x00000010D8143000 (start address = 0x00000010D813B000, size = 69632)
[ 563.072052] HW.Memory \<Error\> core\memory.cpp:operator ():239: Unmapped ReadBlock @ 0x00000010D8144000 (start address = 0x00000010D813B000, size = 69632)
[ 563.072052] HW.Memory \<Error\> core\memory.cpp:operator ():239: Unmapped ReadBlock @ 0x00000010D8145000 (start address = 0x00000010D813B000, size = 69632)
[ 563.072053] HW.Memory \<Error\> core\memory.cpp:operator ():239: Unmapped ReadBlock @ 0x00000010D8146000 (start address = 0x00000010D813B000, size = 69632)
[ 563.072053] HW.Memory \<Error\> core\memory.cpp:operator ():239: Unmapped ReadBlock @ 0x00000010D8147000 (start address = 0x00000010D813B000, size = 69632)
[ 563.072054] HW.Memory \<Error\> core\memory.cpp:operator ():239: Unmapped ReadBlock @ 0x00000010D8148000 (start address = 0x00000010D813B000, size = 69632)
[ 563.072054] HW.Memory \<Error\> core\memory.cpp:operator ():239: Unmapped ReadBlock @ 0x00000010D8149000 (start address = 0x00000010D813B000, size = 69632)
[ 563.072055] HW.Memory \<Error\> core\memory.cpp:operator ():239: Unmapped ReadBlock @ 0x00000010D814A000 (start address = 0x00000010D813B000, size = 69632)
[ 563.072056] HW.Memory \<Error\> core\memory.cpp:operator ():239: Unmapped ReadBlock @ 0x00000010D814B000 (start address = 0x00000010D813B000, size = 69632)


I see that there is a new system update 16.0.3 https://switchbrew.org/wiki/16.0.3. It mentions:

> FS
Mutex lock and unlock were added to some SaveData functions which were previously missing them.
In particular, this is nn::fs::save::SaveDataFileSystemCore::HasOpenedFiles and nn::fs::save::SaveDataFileSystemCore::HasOpenedEntries.

Are there any correlation between the freezing game and not having this system update?

**Comment by lavinsky** [comment]
@DapperCodErr I don't think so, issue seems to be related with VRAM management and ASTC decoding

**Comment by NightrainsRbx** [comment]
A way of solving this would be fixing a different bug which is the texture glitches when enabling Async ASTC decode, since that option fixes this issue, but introduces another bug, I also think it's much easier to reproduce.

**Comment by liamwhite** [comment]
It seems like there is actually no connection between the ASTC settings and this error. It's consistent on the RX 560 4GB no matter what combination I have, but it's not clear yet why it's happening, at least to me.

**Comment by lavinsky** [comment]
I saw a youtube video of some guy on a RX580 8GB playing the game just fine, VRAM usage was about 5.3GB. https://www.youtube.com/live/DGi6WkMtD7k?feature=share&t=374 (notice menu at 6:14)

Maybe the emulator is freeing up VRAM too agressively on less than 6GB cards and when you bring up any menu it unloads the textures (because there isn't enough VRAM) then loads it again. I even thought #10302 would help, tested it but it does nothing.

I've noticed VRAM usage drops sharply and quickly raises again whenever the issue occurs.

**Comment by R4mb4Z4mb4** [comment]
Hey, I have exactly that problem myself but on a RX 570 card and I can confirm, that almost all other people with this card, also have that "menu problem". However, I can describe things on my side a bit more detailed if that helps.

**Scenario 1**: "decode astc asynch." disabled -> results in pretty much the same freezes, and little stops of the game, like shown in the above video. 

**Scenario 2**: "decode astc asnych." enabled -> results also in freezes and little stops, but the behaviour is significantly different. When enabled, the textures basically blur out/become blank, when interacting with the menu or quick menu, and reload every single time again after closing the menu, which let's the CPU (i5 13400) spike to 100% on ALL CORES for this brief moment, which then causes the game to stutter and sometimes even crash. 

This is how my cpu behaves in scenario 2: https://imgur.com/Xmv41h0 

BEFORE the spikes is me just playing, fighting included, heavy shader building included as well, which doesnt seem to be an issue for my cpu, nor gpu. Stable 30 fps. The SPIKES you see, are me changing weapons and fusing items to an arrow.

This is how my cpu behaves in scenario 1: https://imgur.com/1Id2iCp

What I also noticed: All of those problems are usually not triggered when you really fast enter the menu and leave it again. But if you open it for around a second, then the problems happen. So mass spaming menu button seems to not trigger these issues.

While I noticed this weird behaviour on my CPU, i dont think its actually the cpu thats the problem. It feels almost as if Yuzu in combination with the RX 570 somehow erases things and builds them up again from scratch, which shouldnt be this way, which then lets the CPU do strange things.

Another thing: When  "decode astc asnych." is disabled, both my CPU and GPU seem now to do the opposite. Instead of 100% spikes, both the gpu VRAM and 3D value in the task manager, alongside the cpu, seems to DECREASE into zero usage, before then going up again, resulting in the lag, or sometimes even crashes. 

Here are two log files. First is with decode astc asynch enabled, second is disabled.

[decode ASTC enabled.txt](https://github.com/yuzu-emu/yuzu/files/11473927/decode.ASTC.enabled.txt)
[decode ASTC disabled.txt](https://github.com/yuzu-emu/yuzu/files/11473929/decode.ASTC.disabled.txt)

**Comment by Squall-Leonhart** [comment]
note that decode async is a hack and disabled by default, the astc option being discussed is Accelerate ASTC, on the primary graphics menu as a checkbox.

**Comment by R4mb4Z4mb4** [comment]
I actually wanted to add: "and as for accelerate astc in the main settings", it makes no difference in any of my described scenarios. But as you said it, i tested it again. And i can confirm, it makes no different. Turning it on or off, has no impact on my side. Either I get scenario 1, when decode astc is disabled, or scenario 2, when its enabled. Wether I turn on or off accelerate astc.

**Comment by EgoBizarro** [comment]
For some reason I've only started getting this issue today...
I have a GTX 1070 with 8GB VRAM. Maybe that recent change that doesn't let yuzu use all available VRAM produces this issue (which seems to have happened with low VRAM cards before)?

**Comment by gustakasn0v** [comment]
Before 3591 I experienced this, but Yuzu never crashed, FPS just dropped and stuttered. After 3591 I've consistently experienced crashing when this happens while shaders are being compiled (ie. shaders start compiling so I pause gameplay for a bit by opening the map, wait a few seconds, when I exit the map Yuzu crashes). 

I'm on a Steam Deck. CPU Accuracy unsafe and all optimisations enabled; disk pipeline cache, async GPU emulation, accelerate ASTC texture decoding, reactive flushing, async shader building, fast GPU time, Vulkan pipeline cache ENABLED; everything else on Graphics and Advanced graphics DISABLED; accuracy level high, bilinear window adapting.

Lmk if you'd like me to grab some logs, although there seems to be plenty of info on the thread already

**Comment by EgoBizarro** [comment]
Haw, it seems the freezing issue is not only with menus. For some reason I'm getting it randomly (but very frequently, like, every 20 seconds or so) in shrines (at least in the 'upright' shrine at Kakariko village). I guess it's the same issue because async ASTC decoding helps with it as well (but yeah, graphical glitches).

**Comment by liamwhite** [comment]
The massive lag happens because the garbage collector sees that buffer and image resources are being unreferenced for more than a few dozen frames and deletes them, and then immediately needs them again. If I disable garbage collection there is no lag when menuing, but obviously it explodes the memory usage significantly. I think we could handle this with a change to when GC should run, and LFU instead of LRU. This still doesn't explain:
1. the crashing
2. the graphical glitches resulting from GC runs

**Comment by R4mb4Z4mb4** [comment]
The weird thing with it all is also, that it won't get triggered when you just do it fast enough, meaning opening the menu and closing it. But once you do it for more than, I'd say 0.5 seconds, then both VRAM and 3D rendering utilization drops to the floor and rises back up once you leave the menu, which results in freezes. I also saw that the shared gpu memory is always shortly rising in those moments, even if vram is not fully utilized/used. 

Depending if I have enabled "decode astc textures" (i know its not meant to be used in this case), i either get then also 100% spikes on the cpu alongside with it (when leaving the menu), which results visually also in the textures bluring out and reloading, or if disable that option, i get long freezes, sometimes crashes. Turning "accelerate astc" in main graphic settings on or off, makes no difference.

**Comment by liamwhite** [comment]
ASTC has absolutely nothing to do with the crashes, the underlying issue is likely caused by the garbage collector. Here is it on the RX 560 4GB with GC enabled (which eventually crashes, but I didn't record that part):

https://github.com/yuzu-emu/yuzu/assets/9658600/3c8324ac-3db0-4325-a84e-47d72d0d757d


And now disabled, which never crashes:

https://github.com/yuzu-emu/yuzu/assets/9658600/ba443d29-43f5-4af6-9408-b157484f8266

**Comment by R4mb4Z4mb4** [comment]
I see. Yeah, I mean sounds good if then the freezes are gone, since this menu problem is for me so far, the actual game breaker. While trying to get rid of that, I didn't even bother about depths crashing etc. This is how annoying it is. But you said, this change there would go on cost of exploding vram? and are there any other downsides to GC being disabled?

Btw, the comment from me up there was to the guy before me, before I saw yours. But I got it now, that ASTC plays no role there as you said.

**Comment by lavinsky** [comment]
> ASTC has absolutely nothing to do with the crashes, the underlying issue is likely caused by the garbage collector. Here is it on the RX 560 4GB with GC enabled (which eventually crashes, but I didn't record that part):
> gc.mp4
> 
> And now disabled, which never crashes:
> nogc.mp4

Can't we set the garbage collector to be less agressive? Maybe only purge the textures after X time has passed. This isn't a definitive solution but it helps. What are the other possible solutions when the game wont reference certain textures for several frames?

**Comment by Corkor666** [comment]
Enabling "Decode ASTC textures asynchrounously" fixes the crashes for me, though I get some graphical glitches. Running a 1050 2gb, i3 8100, 16gb ram.

**Comment by liamwhite** [comment]
I am still able to get menuing crashes with asynchronous ASTC enabled, it's just a bit more difficult to experience it.

**Comment by EgoBizarro** [comment]
For some reason it's much better for me today than yesterday. Was there a change to GC? Didn't great anything about it.

**Comment by Amer-xyz** [comment]
In my experience it only crashes when you open and close the menu very fast. If you wait a few seconds after opening before closing it again then chances are you'll be able to go a while without crashing.

**Comment by Deadly143** [comment]
Tested on: gtx 970, gtx 1050ti and gtx 1060 3gb. In all cases freezes occur, at least its not crashing. Still because of this its hard to enjoy the game.

**Comment by R4mb4Z4mb4** [comment]
Are there realistically speaking, any potential solutions for the menu problems in the pipeline, so that it soon "could" be fixed? Or do I, as a RX 570 card owner give up on it and just buy a new GPU? Obviously, I know no one can look into the future and people are working on it. But what is the current stance on it and what can be expected for the next days/weeks? Appreciate your work, Liam!

**Comment by Rusxaker** [comment]
> Enabling "Decode ASTC textures asynchrounously" fixes the menu crashes for me, though I get some graphical glitches. Running a 1050 2gb, i3 8100, 16gb ram.

same for me too

**Comment by smllb** [comment]
Game runs stable but if you open the menu, it's over.
Sadly.
Also, how can i disable the garbage collector?

**Comment by R4mb4Z4mb4** [comment]
> Game runs stable but if you open the menu, it's over.
> Sadly.
> Also, how can i disable the garbage collector?

+ still crashing very quickly in the depths, at least for me on my rx570. Vram maxed out, boom, crash.

**Comment by Squall-Leonhart** [comment]
The only guaranteed way to minimise crashes right now is to have as little as possible consuming vram prior to starting Yuzu, and using a 1x scale only, 

some idea's are bouncing around right now and I have heard one possible way that would involve significant reductions to the vram used by the ASTC textures themselves but there are definitely a number of ways that could be used individually or concertively to take care of it, once concrete means of making sure nothing else is affected detrimentally elsewhere.

Something that may or may not come to Windows is now available in Vulkan, on Linux called VK_EXT_pageable_device_local_memory, this would make the device local handler behave more like OpenGL and D3D12 on low video memory systems allowing the application to page our stale video resources rather than needing to evict them.

**Comment by johnecheah** [comment]
Hi, finally I am able to playing the game without menu freeze, I want to share my settings here, I hope can help someone with the same problem.

my pc spec: CPU: i3; Graphic: AMD RX550 4gb, Memory: 8gb

AMD Driver: [Adrenalin edition 23.3.1](https://drivers.amd.com/drivers/whql-amd-software-adrenalin-edition-23.3.1-win10-win11-mar7.exe)
Latest versions of Microsoft Visual C++: [Download](https://aka.ms/vs/17/release/vc_redist.x64.exe)

Yuzu version 1438

API: Vulkan;
Graphics settings: use disk pipeline cache, use asynchronous GPU emulation & accelerate ASTC texture decoding (Enable/Tick)
Resolution: 1x
Window Adapting Filter: AMD FidelityFX super resolution
Anti-Aliasing method: FXAA

Adv. Graphics: Accuracy level: Normal
Enable Reactive Flushing, use asynchronous shader building (Hack) & use vulkan pipeline cache (Enable/Tick), others disable.
Anisotropic Filtering: 16x

I just tested it yesterday night, I run the game about 30min, no menu freeze, change weapon no freeze, use hand ability no freeze, but not test in the shrines.

**Comment by R4mb4Z4mb4** [comment]
> Hi, finally I am able to playing the game stable without menu freeze, I want to share my settings here, I hope can help someone with the same problem.

Just tested your magic settings with a RX 570. I opened the game, opened the menu once and it directly crashed. Restarted Yuzu, cleared pipeline cache, tried to open the menu and it crashed again, this time by the third time i opened the menu.

So for me, your stuff doesn't work and if im honest, i also wouldnt know why it should work when the issue clearly hasnt been fixed yet.

**Comment by johnecheah** [comment]
> > Hi, finally I am able to playing the game stable without menu freeze, I want to share my settings here, I hope can help someone with the same problem.
> 
> Just tested your magic settings with a RX 570. I opened the game, opened the menu once and it directly crashed. Restarted Yuzu, cleared pipeline cache, tried to open the menu and it crashed again, this time by the third time i opened the menu.
> 
> So for me, your stuff doesn't work and if im honest, i also wouldnt know why it should work when the issue clearly hasnt been fixed yet.
> 
> _[ 76.256441] HW.Memory Error core\memory.cpp:operator ():239: Unmapped ReadBlock @ 0x00000010D7AC6000 (start address = 0x00000010D7AB9000, size = 69632) [ 76.256444] HW.Memory Error core\memory.cpp:operator ():239: Unmapped ReadBlock @ 0x00000010D7AC7000 (start address = 0x00000010D7AB9000, size = 69632) [ 76.256445] HW.Memory Error core\memory.cpp:operator ():239: Unmapped ReadBlock @ 0x00000010D7AC8000 (start address = 0x00000010D7AB9000, size = 69632) [ 76.256447] HW.Memory Error core\memory.cpp:operator ():239: Unmapped ReadBlock @ 0x00000010D7AC9000 (start address = 0x00000010D7AB9000, size = 69632)_

v-sync off, not using any mode, just used update 1.1.0.

AMD Adrenalin Graphic card setting, disable all setting: super resolution, image sharpening, etc.

**Comment by Squall-Leonhart** [comment]
It's not going to be of help because the vram loading is going to be different from one system to the next.

**Comment by Doguinhoo** [comment]
Also have the same issue.
Enabling "Decode ASTC textures asynchronously (Hack)" seems to fix the issue, but introduces graphical bugs.

Build: Early Access 3599
CPU: Intel Core i5-12600K
GPU/Driver: NVIDIA GTX 1060 3GB, driver 530.41.03
RAM: 16GB
OS: Garuda Linux (based on Arch Linux)

**Comment by puglin1234** [comment]
Idk if this information will help with anything, but im having the same issue
Also, Enabling "Decode ASTC textures asynchronously (Hack)" fix the issue, but introduce graphical bugs, and causes crashes most of the time

Build: Early Access 3599 / Mainline 1439
CPU: Intel Core i3-9100F
GPU: Nvidia GTX 1050 2GB, Driver 531.79
RAM: 24GB
OS: Windows 11 Pro 22H2

(Downgrading GPU Drivers doesnt seem to help in any way)

**Comment by R4mb4Z4mb4** [comment]
So I have to ask again a more general question: Is there even a potential solution in the pipeline for those menu problems? Or is there no solution in sight and one better shouldn't wait for it, and instead get a new gpu (when he wants to run totk on yuzu without these menu problems)? 

I don't wanna appear impatient or so, but i'd just like to know what @liamwhite thoughts are on that. I don't have a switch sadly and I'd like to play the game at one point. I can wait ofc a bit more, but if its pointless to wait, then might as well upgrade. 

So would more patience be a good advice in this scenario here or better switching hardware, since there seems to be no good solution for it in the near future?

**Comment by johnecheah** [comment]
> Also have the same issue. Enabling "Decode ASTC textures asynchronously (Hack)" seems to fix the issue, but introduces graphical bugs.
> 
> Build: Early Access 3599 CPU: Intel Core i5-12600K GPU/Driver: NVIDIA GTX 1060 3GB, driver 530.41.03 RAM: 16GB OS: Garuda Linux (based on Arch Linux)

hi, Decode ASTC textures asynchronously (Hack) must disable/untick.

**Comment by DCx7C5** [comment]
![errorlogzelda](https://github.com/yuzu-emu/yuzu/assets/42419559/b547d316-e236-41ba-99d1-1c38fdb88dce)

**Comment by Kroissan** [comment]
Same issue here config:
i5 13500
32g ram
rx6800 (16gb vram so probably isn't vram related ?)
tried with astc decode and async compile disabled too
![image](https://github.com/yuzu-emu/yuzu/assets/66138267/fe72fde2-5a0a-4c22-a0ea-37f5018fdb30)

nerver occured before yuzu 1440, reverted to 1439 for now fixed the issue

**Comment by R4mb4Z4mb4** [comment]
Yep, menu freezes are fully back it seems, now after things have been reversed with the recent updates. On my RX 570 i get menu freezes even in BC1. 

I am still wondering to this day, why when those freezes on the 4gb cards are only vram releated, i always get this behaviour of the cpu during those freezes: https://imgur.com/kq7qDrc

When they don't happen, there no spikes at all on the cpu and its just 6 cores having the load with zero load on the other cores. 

this was the case also prior those changes to the garbage collector. If someone knows why the cpu behaves like that during the menu freezes, let me know.

**Comment by kwakenung** [comment]
I have the same issue. It was working really well before now I can hardly play 30 minutes without it happening (RX 580)

**Comment by kwakenung** [comment]
It seems Zora's Domain/Lanayru is a problematic area.

**Comment by kwakenung** [comment]
Starting a new shader file seems to have helped as well?

**Comment by devmachine1** [comment]
so for me this started in EA 3621, game freezes after exiting menu then comes back to normal

**Comment by SonicGleb** [comment]
So it's a different thing for me, but I think the same issue. Whenever I just teleport to any location or travel somewhere, softlock may happen and the only way out is to restart.

**Comment by gSuuuuP** [comment]
RX570 here.
Tried yesterday with Yuzu last version (not EA), with BC1, I am finally able to go down in Abyss with no issue!
Still got freeze when exiting "big menu" (map, saves), but no more freeze (most of the time, still happen time to time, don't know why) when exiting weapon selection menu or abilities menu. It will be OK for ever if I don't teleport or enter a shrine. Sometime when I teleport somewhere, I have menu freeze, then I TP again, and it's fine.

**Comment by Maleriandro** [comment]
Is there some possibility that this will eventually get fixed?

**Comment by Amer-xyz** [comment]
Unlikely, they have the fix but it causes stability issues in other games.

**Comment by liamwhite** [comment]
Fix was attempted in #10529 but it broke rendering in TOTK due to other bugs in the texture cache. Those need to be fixed before that can be attempted again.

**Comment by R4mb4Z4mb4** [comment]
> Fix was attempted in #10529 but it broke rendering in TOTK due to other bugs in the texture cache. Those need to be fixed before that can be attempted again.

And is there something in the pipeline, such as some ideas what potentially could fix those issues that need to dealt with prior to putting the hands again on the menu freezes? Or is it currently basically a "no idea anymore" point?

It's just so sad with this menu nonsense. Apart from it, it's really playable, even on my RX 570. But the menu issues destroy everything, sadly.

**Comment by gustafla** [comment]
I'm sure that the people who have been following this crash issue know it already, but I have to remind to stop running (outdated) mods. I thought my crashes were caused by this, but it was actually an FPS mod.

**Comment by gustafla** [comment]
Yes. Sorry for unclear wording.

**Comment by johnecheah** [comment]
I am using RX550 and yuzu ea 3891, follow the setting from this [https://yuzu-emu.org/entry/yuzu-progress-report-aug-2023/](url), the menu freeze problem solved.

the ASTC recompression method must be set to Uncompressed, and Enable asynchronous presentation should be enabled if you’re using Vulkan. Both options can be found in Emulation > Configure… > Graphics > Advanced.

**Comment by Whosyourdaddy999999** [comment]
I just disabled "Use asynchronous GPU emulation" and no more menu lag for me. Is it bad to disable this setting?

**Comment by edymelendez** [comment]
> I just disabled "Use asynchronous GPU emulation" and no more menu lag for me. Is it bad to disable this setting?

Tried it in a fight where I had a bunch of freezing after switching between bows/weapons. Reloaded the fight to recreate the issue after disabling this setting. No more freezing! Worth trying  ~Steam Deck User

**Comment by edymelendez** [comment]
> I just disabled "Use asynchronous GPU emulation" and no more menu lag for me. Is it bad to disable this setting?

Worked for me! ~Steam Deck user
